### ISSUE TITLE
[ally]: Combobox Default & with Layer - VO focus order

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#16045]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

> What did you expect to happen?
1. Focus should be set back on the triggering control after selection
2. Redundant swipe focus should be removed

> What happened instead?

Following behavior is observed in VO iPhone. 
1. After selecting an option from the dropdown, the Screen reader focus is not set back on the triggering control. 
2. After selection, When the user tries to navigate back, it takes 5 blank swipe stopped to reach the triggering control. Note: This happens each time when the user selects an option.

> User impact
When the focus is not set on a logical point in the page after a user action, it creates a great deal of confusion. The whole process of navigating back to the relevant element can be tedious. note this defect mainly impacts keyboard-only users and those with cognitive disabilities. 

> What WCAG 2.1 checkpoint does the issue violate?
2.4.3 Focus Order 

## Steps to reproduce the issue

Default: https://react.carbondesignsystem.com/?path=/story/components-combobox--default
With layer: https://react.carbondesignsystem.com/?path=/story/components-combobox--with-layer

## Additional information

- Screenshots or code
- Notes
